svir/dialogs/load_gmf_data_as_layer_dialog.py

```diff
@@ -0,0 +1,140 @@
+                                  log_msg,
+                                  )
+from svir.utilities.shared import DEBUG
+
+
+class LoadGmfDataAsLayerDialog(LoadOutputAsLayerDialog):
+    """
+    Modal dialog to load gmf_data from an oq-engine output, as layer
+    """
+
+    def __init__(self, iface, output_type='gmf_data', path=None, mode=None):
+        assert(output_type == 'gmf_data')
+        LoadOutputAsLayerDialog.__init__(self, iface, output_type, path, mode)
+        self.setWindowTitle(
+            'Load ground motion fields from NPZ, as layer')
+        self.create_load_selected_only_ckb()
+        self.create_rlz_selector()
+        self.create_imt_selector()
+        self.create_eid_selector()
+        if self.path:
+            self.npz_file = numpy.load(self.path, 'r')
+            self.populate_out_dep_widgets()
+        self.adjustSize()
+        self.set_ok_button()
+
+    def set_ok_button(self):
+        self.ok_button.setEnabled(
+            bool(self.path)
+            and self.imt_cbx.currentIndex() != -1)
+
+    def on_rlz_changed(self):
+        self.dataset = self.npz_file[self.rlz_cbx.currentText()]
+        imts = self.dataset.dtype.names[2:]
+        self.imt_cbx.clear()
+        self.imt_cbx.setEnabled(True)
+        self.imt_cbx.addItems(imts)
+        self.set_ok_button()
+
+    def on_imt_changed(self):
+        imt = self.imt_cbx.currentText()
+        min_eid = 0
+        max_eid = (self.dataset[imt].shape[1] - 1)
+        self.eid_sbx.cleanText()
+        self.eid_sbx.setEnabled(True)
+        self.eid_lbl.setText(
+            'Event ID (used for default styling) (range %d-%d)' % (
+                min_eid, max_eid))
+        self.eid_sbx.setRange(min_eid, max_eid)
+        self.set_ok_button()
+
+    def populate_rlz_cbx(self):
+        self.rlzs = [item[0] for item in self.npz_file.items()]
+        self.rlz_cbx.clear()
+        self.rlz_cbx.setEnabled(True)
+        # self.rlz_cbx.addItem('All')
+        self.rlz_cbx.addItems(self.rlzs)
+
+    def load_from_npz(self):
+        for rlz in self.rlzs:
+            if (self.load_selected_only_ckb.isChecked()
+                    and rlz != self.rlz_cbx.currentText()):
+                continue
+            with WaitCursorManager('Creating layer for realization "%s"...'
+                                   % rlz, self.iface):
+                self.build_layer(rlz)
+                self.style_maps()
+        if self.npz_file is not None:
+            self.npz_file.close()
+
+    def build_layer_name(self, rlz, **kwargs):
+        self.imt = self.imt_cbx.currentText()
+        self.eid = self.eid_sbx.value()
+        self.default_field_name = '%s-%s' % (self.imt, self.eid)
+        # layer_name = "gmf_data_%s_eid-%s" % (rlz, self.eid)
+        layer_name = "scenario_damage_gmfs_%s_eid-%s" % (rlz, self.eid)
+        return layer_name
+
+    def get_field_names(self, **kwargs):
+        field_names = list(self.dataset.dtype.names)
```

Comment:
Yes, I need a list instead of a tuple, because afterwards I need to tweak that list, while adding fields to the layer I am creating. It is because I have to keep track of how field names are laundered (for instance to be compliant with shapefiles constraints).